plutus-metatheory/src/Builtin/Signature.lagda.md

```diff
@@ -123,37 +133,151 @@ module FromSig (Ctx : Set)
     ♯2* (con (pair x y)) = mkTyCon  (T2.pair (♯2* x) (♯2* y))
     ♯2* (con pdata) = mkTyCon  T2.pdata
 
-    {- `sig2type-aux` takes a list of arguments and a result type, and produces
-        a function that takes all arguments and returns the result type.
+    -- The empty context
+    ∅ : Ctx 
+    ∅ = nat2Ctx 0
+```
+
+### Types originating from a Signature
 
-       sig2type-aux [ b₁ , b₂ , ... ,bₙ ] tᵣ = t₁ ⇒ t₂ ⇒ ... ⇒ tₙ ⇒ tᵣ
-          where tᵢ = ♯2* bᵢ
-    -}
-    sig2type-aux : ∀{n} 
-              → Args n 
-              → Ty (nat2Ctx n) → Ty (nat2Ctx n)
-    sig2type-aux xs = foldr (λ A xs res → (♯2* A) ⇒ (xs res)) (λ A res → ♯2* A ⇒ res) xs
+The types produced by a signature have a particular form: possibly 
+some Π applications and then at least one function argument.
 
-    -- take a type and close its type variables using Π
-    sig2type-aux2 : ∀{n} → Ty (nat2Ctx n) → Ty (nat2Ctx 0)
-    sig2type-aux2 {zero} t = t
-    sig2type-aux2 {suc n} t = sig2type-aux2 {n} (Π t)
+We define a predicate for concrete types of that shape as a datatype 
+indexed by the concrete types
+
+```
+    -- an : number of arguments to be added to the type 
+    -- am : number of arguments expected
+    -- at : total number of arguments
+{-
+    data SigTy⇒ : ∀{n an am at} → an ∔ am ≣ at → Ty (nat2Ctx n) → Set where
+       bresult  : ∀{n at}
```

Comment:
why `bresult`?

Reply:
builtin result. And `result` was used in the signature.